### DISCUSSION TITLE
How best to ingest emails with attached documents?

### DISCUSSION THREAD

**Discussion by nileshtrivedi** [microsoft/graphrag#486] [Q&A]
I was hoping to create a knowledge graph from my email inbox. But emails often have attachments and separating the email body from attachment files would lose crucial links.

How can I go about ingesting emails into a knowledge graph properly?

**Comment by silhouettehustler** [comment]
You'll have to pre-process your email data in a way that links your email content with the relevant attachments, obviously you'll need to extract the attachment's content as well as the email content and bind it all together in your input text file that'll get indexed by GraphRAG's indexing engine.

 Something like below:

####

Emails:

-Subject: Test email
 From: test@gmail.com
 To: me@gmail.com
 Content: This is a test email
 Attachments: This is my attachment content from PDF1; There's more attachment content from PDF2

-Subject: Test email 2
 From: test2@gmail.com
 To: me2@gmail.com
 Content: This is a test email 2
 Attachments: This is my attachment content from PDF3; There's more attachment content from PDF4

####
 
 Obviously this is just an example format of what your input text file could look like , I'm sure you can improve on this and add more relevant metadata, but ultimately you wanna end up with a text file having all your email information in there.
